src/prost/src/lib.rs

```diff
@@ -220,7 +221,11 @@ impl stream_plan::MaterializeNode {
 // Encapsulating the use of parallelism.
 impl common::WorkerNode {
     pub fn parallelism(&self) -> usize {
-        self.parallelism as usize
+        assert_eq!(self.r#type(), WorkerType::ComputeNode);
+        self.property
+            .as_ref()
+            .expect("property should be exist")
+            .parallelism as usize
```

Comment:
This will panic in some places:
- diagnose in dashboard
- `show cluster`

Can you examine the usages of `worker.parallelism()`? (Or you may change `list_all_nodes()` calls to `list_cn()`)